pallets/thea-executor/src/lib.rs

```diff
@@ -535,16 +576,13 @@ pub mod pallet {
 		}
 
 		#[transactional]
-		pub fn execute_deposit(
-			deposit: Deposit<T::AccountId>,
-			recipient: &T::AccountId,
-		) -> Result<(), DispatchError> {
+		pub fn execute_deposit(deposit: Deposit<T::AccountId>) -> Result<(), DispatchError> {
 			// Get the metadata
 			let metadata =
 				<Metadata<T>>::get(deposit.asset_id).ok_or(Error::<T>::AssetNotRegistered)?;
 			let deposit_amount = deposit.amount_in_native_decimals(metadata); // Convert the decimals configured in metadata
 
-			if !frame_system::Pallet::<T>::account_exists(recipient) {
+			if !frame_system::Pallet::<T>::account_exists(&deposit.recipient) {
```

Comment:
If the account doesn't exist, can't we add this to a pallet account which is guaranteed to exist? this way, if any user makes the mistake of calling it without ed, the funds can still be given back to the user via governance if needed. it should be better than making the funds "vanish"

Reply:
@zktony What do you think of sending it to pallet account ? We may be need an governance-origin extrinsic to move the funds out but that's okay.

Reply:
yeah .. we can do that

Reply:
Resolved.